File: output/CodeGen/avatar/Java/Python/atcoder_ABC107_C.py
def main ( ):
    N = int(input())
    K = int(input())
    S = [int(i) for i in input().split()]
    temp = 0
    ans = 1000000000
    if K == 1:
        for i in range(0, N):
            temp = S[i]
            ans = min(abs(temp), ans)
        print(ans)
    else:
        if N - K!= 0:
            for i in range(0, N - K + 1):
                min = S[i]
                max = S[i + K - 1]
                if min < 0 and max > 0:
                    temp = 2 * (- min) + max
                else:
                    temp = max if max > 0 else min
                ans = min(ans, temp)
            print(ans)
        else:
            min = S[0]
            max = S[N - 1]
            if min < 0 and max > 0:
                print(2 * (- min) + max)
            else:
                print(max if max > 0 else min)

if __name__ == "__main__":
    main()

File: output/ChatGPT/avatar/Python/Java/atcoder_ABC107_C.java
import java.util.Scanner;

public class atcoder_ABC107_C {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int N = input.nextInt();
        int K = input.nextInt();
        int[] lst = new int[N];
        for (int i = 0; i < N; i++) {
            lst[i] = input.nextInt();
        }
        int[] lst_p = new int[N];
        int[] lst_m = new int[N];
        int p = 0;
        int m = 0;
        int x = 0;
        int y = lst_m.length - 1;
        int lastx = lst_p.length;
        int lasty = lst_m.length;

        int result = check(K, lst_p, lst_m, p, m, x, y, lastx, lasty);
        System.out.println(result);
    }

    public static int check(int k, int[] lst_p, int[] lst_m, int p, int m, int x, int y, int lastx, int lasty) {
        int ans = (int) Math.pow(10, 9);
        for (int j = 0; j <= k; j++) {
            if (j > lastx || k - j > lasty) {
                continue;
            } else {
                if (j == 0) {
                    p = 0;
                } else {
                    p = lst_p[j - 1];
                }
                if (k - j == 0) {
                    m = 0;
                } else {
                    m = lst_m[lasty - (k - j)];
                }
                ans = Math.min(ans, 2 * p + Math.abs(m));
                ans = Math.min(ans, p + 2 * Math.abs(m));
            }
        }
        return ans;
    }
}